source/rock/middle/Addon.ooc

```diff
@@ -155,9 +156,16 @@ Addon: class extends Node {
             }
 
             if (call suggest(fDecl, res, trail)) {
-                if (fDecl hasThis() && !call getExpr()) {
-                    // add `this` if needed.
-                    call setExpr(VariableAccess new("this", call token))
+                if (!call getExpr()) {
+                    if (fDecl hasThis()) {
+                        // add `this` if needed.
+                        call setExpr(VariableAccess new("this", call token))
+                    }
+                    // Not yet tested
```

Comment:
I just have to assume that it does not give false detection until someone uses addons.